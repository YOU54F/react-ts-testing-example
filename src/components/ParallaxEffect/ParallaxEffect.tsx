import * as React from 'react';

export interface ParallaxEffectProps {
  parallaxBackgroundImage?: string;
  parallaxText?: string;
  parallaxTextBackground?: string;
}

const handleParallaxTextScroll = () => {
  const x = window.scrollY;
  const parallaxText = document.getElementById('parallax-text');
  if (parallaxText) {
    parallaxText.style.backgroundPosition =
      '100% ' + -x / 1 + 'px' + ', 0%  0%, center top';
  }
};

window.onscroll = () => {
  handleParallaxTextScroll();
};

export class ParallaxEffect extends React.Component<ParallaxEffectProps, {}> {
  public render() {
    return (
      <div className='parallax-effect-container'>
        {this.props.parallaxBackgroundImage && (
          <div
            className='parallax-image'
            style={{
              backgroundImage: 'url(' + this.props.parallaxBackgroundImage + ')'
            }}
          />
        )}
        {this.props.parallaxText && (
          <div className='parallax-text-container'>
            <span
              className='parallax-text'
              id='parallax-text'
              style={{
                backgroundImage:
                  'url(' + this.props.parallaxTextBackground + ')'
              }}>
              {this.props.parallaxText}
            </span>
          </div>
        )}
      </div>
    );
  }
}
