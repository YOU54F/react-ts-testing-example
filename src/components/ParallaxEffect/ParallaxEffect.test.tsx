import * as React from 'react';
import renderer from 'react-test-renderer';
import { ParallaxEffect } from './ParallaxEffect';

describe('Parallax Effect', () => {
  it('can render the parallax effect', () => {
    const tree = renderer
      .create(
        <ParallaxEffect
          parallaxBackgroundImage='https://66.media.tumblr.com/6d884fe09d90b1bac096df349d7b7b38/tumblr_orxj4oc6AH1uu9fqho1_500.gif'
          parallaxText='An example website'
          parallaxTextBackground='https://66.media.tumblr.com/6d884fe09d90b1bac096df349d7b7b38/tumblr_orxj4oc6AH1uu9fqho1_500.gif'
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
