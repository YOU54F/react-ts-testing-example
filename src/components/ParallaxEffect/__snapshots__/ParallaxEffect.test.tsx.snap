// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parallax Effect can render the parallax effect 1`] = `
<div
  className="parallax-effect-container"
>
  <div
    className="parallax-image"
    style={
      Object {
        "backgroundImage": "url(https://66.media.tumblr.com/6d884fe09d90b1bac096df349d7b7b38/tumblr_orxj4oc6AH1uu9fqho1_500.gif)",
      }
    }
  />
  <div
    className="parallax-text-container"
  >
    <span
      className="parallax-text"
      id="parallax-text"
      style={
        Object {
          "backgroundImage": "url(https://66.media.tumblr.com/6d884fe09d90b1bac096df349d7b7b38/tumblr_orxj4oc6AH1uu9fqho1_500.gif)",
        }
      }
    >
      An example website
    </span>
  </div>
</div>
`;
